packages/core/src/browser/pageExitObservable.ts

```diff
@@ -1,10 +1,12 @@
+import { isExperimentalFeatureEnabled } from '../domain/configuration'
 import { Observable } from '../tools/observable'
-import { includes, objectValues } from '../tools/utils'
-import { addEventListener, addEventListeners, DOM_EVENT } from './addEventListener'
+import { includes, noop, objectValues } from '../tools/utils'
+import { addEventListeners, addEventListener, DOM_EVENT } from './addEventListener'
 
 export const PageExitReason = {
   HIDDEN: 'visibility_hidden',
   UNLOADING: 'before_unload',
+  PAGEHIDE: 'pagehide',
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏At first I wanted to suggest using `PAGE_HIDE: "page_hide"`, but then I'm a bit mitigated. The `PageExitReason` goal was to abstract a bit over event names (which are implementation detail) and to have two variants, one as a "possible exit" where we just want to flush things and the other as a "definitive exit" where we also want to end things.

Now we have four variants and I'm not sure what we want/need to do with each of them. Maybe it could be clarified with something like this:

```ts
interface PageExitEvent {
  definitive: boolean, // whether we should end things or not
  reason: PageExitReason,
}

```

Reply:
To be honest, I have the same filling looking at this observable. I do agree that with that many events, all triggered in different page lifecycle state change, the `exit` naming is unclear. Maybe this observable as too many responsibilities.
I prefer to think of it as a convenient way to share logic for batch and segment flush.

Maybe we could:
- Use this observable only for the batch and segment flush and chose a better name
- No use it to end view and just use a normal window event listener on pagehide

Finally:
To be sure this PR make sense: I would prefer to move forward with this implementation to investigate the `pagehide` impact. Wdyt?

Reply:
> Your suggestion could make sense but to me it adds another level of complexity where we want to make things easier.

On the contrary, I would argue that it makes things easier as we would have 2 values to consider instead of 4 values (boolean vs enum values).

In your PR, you did not adjust trackViews [here](https://github.com/DataDog/browser-sdk/blob/a483e80b210ef0ef1fa918d7557108f1570b382a/packages/rum-core/src/domain/rumEventsCollection/view/trackViews.ts#L132) to end the view on `pagehide`, so we will lose final view updates. Having a clear boolean instead of relying on the reason value could have been less error-prone.

>  To be sure this PR make sense: I would prefer to move forward with this implementation to investigate the pagehide impact. Wdyt?

I'm fine with keeping the large enum for now, since it is temporary.

Reply:
> In your PR, you did not adjust trackViews [here](https://github.com/DataDog/browser-sdk/blob/a483e80b210ef0ef1fa918d7557108f1570b382a/packages/rum-core/src/domain/rumEventsCollection/view/trackViews.ts#L132) to end the view on pagehide
Good catch 🙂 

> On the contrary, I would argue that it makes things easier as we would have 2 values to consider instead of 4 values (boolean vs enum values).
It might be just me but since we would keep the pageExitReason, adding this boolean would make more combination to understand 2 * number of reasons. Of course not all these combinations would exist but PageExitEvent don't explicitly tells it.

Reply:
> the exit naming is unclear

agree

> one as a "possible exit" where we just want to flush things and the other as a "definitive exit" where we also want to end things

I'm wondering if we could align our concepts with the ones from [page lifecycle api](https://developer.chrome.com/blog/page-lifecycle-api/) and then have something like:

```
PageLifecycleEvent {
  state: 'hidden' | 'frozen' | 'terminated',
  event: 'visibilitychange' | 'pagehide' | 'beforeunload' | 'freeze'
}
```

With that we could:
- listen for `hidden` | `frozen` | `terminated` to flush batches/segments.
- listen for `terminated` to end the views
- use the event names as reason for monitoring purposes

would it make sense?

Reply:
Good idea! I don't want to mix it with this PR so I'll create a followup task

Reply:
I don't think it's particullary simpler than a boolean, and I still think the state and event are implementation details.